File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.cpp
class Solution {
public:
    int minimumPushes(string word) {
        unordered_map<char, int> dict;
        for (char letter : word)
            dict[letter]++;
        vector<pair<char, int>> helper(dict.begin(), dict.end());
        sort(helper.begin(), helper.end(), [](pair<char, int>& lhs, pair<char, int>& rhs){
            return lhs.second > rhs.second;
        });
        int answer = 0;
        int occupied = 0;
        for (pair<char, int> elem : helper){
            answer += (occupied / 8 + 1) * elem.second;
            occupied++;
        }
        return answer;
    }
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        mp = [0 for i in range(ord('z') - ord('a') + 1)]
        for letter in word:
            index = ord(letter) - ord('a')
            mp[index] += 1
        mp.sort(reverse=True)

        answer = 0
        occuried = 0
        for frequince in mp:
            answer += (occuried // 8 + 1) * frequince
            occuried += 1

        return answer 
        